Modules/Filtering/ImageStatistics/include/itkLabelOverlapMeasuresImageFilter.hxx

```diff
@@ -120,28 +120,23 @@ LabelOverlapMeasuresImageFilter<TLabelImage>::ThreadedStreamedGenerateData(const
   // local copy, this thread may do multiple merges.
   while (true)
   {
-
+    MapType tomerge{};
     {
-      std::unique_lock<std::mutex> lock(m_Mutex);
+      const std::lock_guard<std::mutex> lockGuard(m_Mutex);
 
       if (m_LabelSetMeasures.empty())
       {
         swap(m_LabelSetMeasures, localStatistics);
         break;
       }
-      else
-      {
-        // copy the output map to thread local storage
```

Comment:
Note that this PR proposes to remove the term "thread local storage" here, as I think that that term has a specific meaning (https://en.wikipedia.org/wiki/Thread-local_storage) that is not entirely applicable to the local `tomerge` variable in this case.